eventmesh/ApiServerEventMesh.js

```diff
@@ -0,0 +1,411 @@
+        .namespaces(CONST.APISERVER.NAMESPACE)[resourceType](resourceId).patch({
+          body: delta
+        }));
+  }
+
+  deleteLockResource(resourceName, resourceType, resourceId) {
+    return this.deleteResource(resourceName, resourceType, resourceId)
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  updateResource(resourceName, resourceType, resourceId, delta) {
+    return this.patchResource(resourceName, resourceType, resourceId, delta)
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  getLockResourceOptions(resourceName, resourceType, resourceId) {
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver.apis[`${resourceName}.${CONST.APISERVER.HOSTNAME}`][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[resourceType](resourceId).get())
+      .then(resource => {
+        return resource.body.spec.options;
+      })
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  getResource(resourceName, resourceType, resourceId) {
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver.apis[`${resourceName}.${CONST.APISERVER.HOSTNAME}`][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[resourceType](resourceId).get())
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  createDeploymentResource(resourceType, resourceId, val) {
+    const opts = {
+      operationId: resourceId,
+      resourceId: resourceId,
+      operationName: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT,
+      operationType: CONST.APISERVER.RESOURCE_NAMES.DIRECTOR,
+      value: val
+    };
+    return this.createOperationResource(opts);
+  }
+
+  updateResourceState(resourceType, resourceId, stateValue) {
+    const opts = {
+      operationId: resourceId,
+      resourceId: resourceId,
+      operationName: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT,
+      operationType: resourceType,
+      stateValue: stateValue
+    };
+    return this.updateOperationState(opts);
+  }
+
+  getResourceState(resourceType, resourceId) {
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis[`${CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT}.${CONST.APISERVER.HOSTNAME}`][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[resourceType](resourceId)
+        .get())
+      .then(json => json.body.status.state)
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  /**
+   * Craete Resource in Apiserver with the opts
+   * @param {string} opts.resourceId
+   * @param {string} opts.operationName
+   * @param {string} opts.operationType
+   * @param {string} opts.operationId
+   * @param {Object} opts.value
+   */
+  createOperationResource(opts) {
+    logger.info('Creating resource with options:', opts.value);
```

Comment:
Can be changed to `Creating resource ${opts.metadata.name} with options:', opts.value`

Reply:
Done